### PR TITLE
Add `genBoundedEnum`

### PR DESCRIPTION
**Pull request by garyb**

### CODE REVIEW

File: src/Data/Enum/Gen.purs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Data.Enum (class BoundedEnum, fromEnum, toEnum)
+import Partial.Unsafe (unsafePartial)
+import Data.Maybe (fromJust)
+
+-- | Create a random generator for a finite enumeration. `toEnum i` must be
+-- | well-behaved: it must return a `Just` value for all `Int`s between
+-- | `fromEnum bottom` and `fromEnum top`.
+genBoundedEnum :: forall m a. MonadGen m => BoundedEnum a => m a
+genBoundedEnum =
+  unsafePartial fromJust <<< toEnum
```
Comment (paf31) [inline src/Data/Enum/Gen.purs]: This `unsafePartial` is only safe if the `Enum` instance is valid, I think.

Another option is `oneOf (bottom .. top)`, or something along those lines, right?

Comment (garyb) [inline src/Data/Enum/Gen.purs]: This generator I lifted from QuickCheck actually, it was recently added there, so it suffers from any of the same problems that that one has.

Can you explain what you're thinking with `oneOf (bottom .. top)`, I'm not sure I follow?

Comment (garyb) [inline src/Data/Enum/Gen.purs]: It does say in the comment that `toEnum` must be well behaved too :wink:

Comment (paf31) [inline src/Data/Enum/Gen.purs]: By `bottom .. top`, I meant to say `[enumFromTo](https://pursuit.purescript.org/packages/purescript-enums/3.0.0/docs/Data.Enum#v:enumFromTo) bottom top`, which must be safe, I think.

Comment (garyb) [inline src/Data/Enum/Gen.purs]: Ahh yeah, that'd work. It'll be kinda expensive if used for large enums, but can revisit if we have a case where it's obviously terrible. :)

**Comment by garyb** [comment]
How's this? A slight bias to `bottom` now, but necessary without adding dependencies or FFI something, since there's no `array`/`list` dependency even transitively.

**Comment by paf31** [comment]
👍 Thanks!

**Pull request merged**